opcua/common/manage_nodes.py

```diff
@@ -6,22 +6,22 @@
 
 
 def _parse_add_args(*args):
-    nodeid = ua.NodeId()
-    qname = ua.QualifiedName()
-    if isinstance(args[0], ua.NodeId):
-        nodeid = args[0]
-    elif isinstance(args[0], str):
-        nodeid = ua.NodeId.from_string(args[0])
-
-    if isinstance(args[1], ua.QualifiedName):
-        qname = args[1]
-    elif isinstance(args[1], str):
-        if isinstance(args[0], int):
+    try:
+        if isinstance(args[0], ua.NodeId):
+            nodeid = args[0]
+            qname = args[1]
```

Comment:
That's a point...

On Tue, Mar 29, 2016, 18:00 Denis Štogl <notifications@github.com> wrote:

> In opcua/common/manage_nodes.py
> <https://github.com/FreeOpcUa/python-opcua/pull/150#discussion_r57749547>:
>
> > @@ -6,22 +6,22 @@
> >
> >
> >  def _parse_add_args(*args):
> > -    nodeid = ua.NodeId()
> > -    qname = ua.QualifiedName()
> > -    if isinstance(args[0], ua.NodeId):
> > -        nodeid = args[0]
> > -    elif isinstance(args[0], str):
> > -        nodeid = ua.NodeId.from_string(args[0])
> > -
> > -    if isinstance(args[1], ua.QualifiedName):
> > -        qname = args[1]
> > -    elif isinstance(args[1], str):
> > -        if isinstance(args[0], int):
> > +    try:
> > +        if isinstance(args[0], ua.NodeId):
> > +            nodeid = args[0]
> > +            qname = args[1]
>
> OK, but we have more flexibility, Because NodeId can be String and and
> qname QualifiedName object, which is now not possible....
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/FreeOpcUa/python-opcua/pull/150/files/2119fc7f87c32313e9cbabd769d83c818aeed1b2#r57749547>
>